jetty-core/jetty-rewrite/src/main/java/org/eclipse/jetty/rewrite/handler/ResponseHeaderRegexRule.java

```diff
@@ -0,0 +1,138 @@
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.rewrite.handler;
+
+import java.io.IOException;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import org.eclipse.jetty.http.HttpStatus;
+import org.eclipse.jetty.server.Response;
+import org.eclipse.jetty.util.Callback;
+import org.eclipse.jetty.util.StringUtil;
+
+/**
+ * Sends the response code whenever the rule finds a match.
+ */
+public class ResponseHeaderRegexRule extends Rule
+{
+    private String _headerName;
+    private Pattern _headerRegex;
+    private int _code;
+    private String _message;
+
+    public ResponseHeaderRegexRule()
+    {
+    }
+
+    @Override
+    public boolean isTerminating()
+    {
+        return true;
+    }
+
+    public String getHeaderName()
+    {
+        return _headerName;
+    }
+
+    public void setHeaderName(String header)
+    {
+        _headerName = header;
+    }
+
+    public String getHeaderRegex()
+    {
+        return _headerRegex == null ? null : _headerRegex.pattern();
+    }
+
+    public void setHeaderRegex(String headerRegex)
+    {
+        _headerRegex = headerRegex == null ? null : Pattern.compile(headerRegex);
+    }
+
+    public int getCode()
+    {
+        return _code;
+    }
+
+    /**
+     * Set the response code.
+     * @param code the response code
+     */
+    public void setCode(int code)
+    {
+        _code = code;
+    }
+
+    public String getMessage()
+    {
+        return _message;
+    }
+
+    /**
+     * <p>Sets the message for the response body (if the response code may have a body).</p>
+     *
+     * @param message the response message
+     */
+    public void setMessage(String message)
+    {
+        _message = message;
+    }
+
+    @Override
+    public Handler matchAndApply(Handler input) throws IOException
+    {
+        String value = input.getHeaders().get(getHeaderName());
+        if (value == null)
+            return null;
+        if (_headerRegex == null)
+            return apply(input, value);
+        Matcher matcher = _headerRegex.matcher(value);
+        if (matcher.matches())
+            return apply(input, value);
+        return null;
+    }
+
+    public Handler apply(Handler input, String value) throws IOException
+    {
+        if (getCode() < HttpStatus.CONTINUE_100)
+            return null;
+
+        return new Handler(input)
+        {
+            @Override
+            protected boolean handle(Response response, Callback callback)
+            {
+                String message = getMessage();
+                if (StringUtil.isBlank(message))
+                {
+                    response.setStatus(getCode());
+                    callback.succeeded();
+                }
+                else
+                {
+                    Response.writeError(this, response, callback, getCode(), message);
+                }
```

Comment:
Coalesce all of this into a single `Response.writeError()`, as the `message` parameter may be `null`.

Reply:
Ok. This construct was also based on InvalidURIRule.java and ResponsePatternRule.java, I will update to match current recommendations, thanks.

What's the project's preferred standard on accessing private class variables within the class? I noticed some of the files always use the accessor getX() function and others just directly access the private _x variable.

Reply:
We have no rule, but you did right.